### ISSUE TITLE
Yoda comparation not complaining for this case

### ISSUE THREAD

**Issue by tatitati** [djoos/Symfony-coding-standard#113]
I know that YODA comparation tries to avoid a bad assignation to a variable. But also is an standard saying that the expected value should be on the left (first) when comparing.

This said, in my case I can do this without simple variables and phpcs doesn't complain:

```php
if ($e->getErrorCode() == 2006) {
       ....
}
```
I think in this case also should complain. This makes the code more standard having all the expected values on the left.

**Comment by straticJeff** [comment]
I think the answer is in your post - Yoda style helps avoid unintended dangerous _variable_ assignments. But your example is a comparison against a _function call_, which you cannot accidentally assign in PHP. The Yoda style rule is there to help devs proactively defend against typos that are difficult to spot and debug, but as your example can't be dangerous, I don't think it should apply in this case.

**Comment by djoos** [comment]
Closing this one off for now - happy to revisit the discussion if needed in the future!

**Comment by outcomer** [comment]
@djoos Hey!
What about this?
`if ($args['type'] == 1) {....}`

YODA rule also would not trigger in this case due to shiff logic
`if (T_VARIABLE === $tokens[$comparison - 2]['code']
                && T_VARIABLE !== $tokens[$comparison + 2]['code']
 ) {`

For this `if ($args == 1) {....}` YODA rule WILL works.
